ros/src/tl_detector/tl_detector.py

```diff
@@ -131,23 +144,36 @@ def process_traffic_lights(self):
         Returns:
             int: index of waypoint closes to the upcoming stop line for a traffic light (-1 if none exists)
             int: ID of traffic light color (specified in styx_msgs/TrafficLight)
-
         """
-        light = None
+        closest_light = None
+        line_wp_idx = None
 
         # List of positions that correspond to the line to stop in front of for a given intersection
         stop_line_positions = self.config['stop_line_positions']
-        if(self.pose):
-            car_position = self.get_closest_waypoint(self.pose.pose)
-
-        #TODO find the closest visible traffic light (if one exists)
+        if self.pose:
+            position = self.pose.pose.position
+            car_wp_idx = self.get_closest_waypoint(position.x, position.y)
+            # TODO find the closest visible traffic light (if one exists)
+            diff = len(self.waypoints.waypoints)
+            for i, light in enumerate(self.lights):
+                # Get sto line waypoint index
+                line = stop_line_positions[i]
+                temp_wp_idx = self.get_closest_waypoint(line[0], line[1])
```

Comment:
@ank700 this is where the closest traffic light is found.
Or have you found an error?

Reply:
I found this section. Some parts of the tl_detector.py were missing in my fork and I got confused. I have cloned the whole code this time from the MD/simple_tl_controller branch.

Reply:
Merging.